src/simplewallet/simplewallet.cpp

```diff
@@ -3268,6 +3281,18 @@ bool simple_wallet::donate(const std::vector<std::string> &args_)
     payment_id_str = local_args.back();
     local_args.pop_back();
   }
+  // check priority
+  if (local_args.size() > 1)
+  {
+    auto priority_pos = std::find(
+      allowed_priority_strings.begin(),
+      allowed_priority_strings.end(),
+      local_args[0]);
+    if(priority_pos != allowed_priority_strings.end()) {
+      priority_str = local_args[0];
+      local_args.erase(local_args.begin());
+    }
+  }
```

Comment:
That looks like it should be a helper function

Reply:
OK, now fixed, does it look good?